app/main/routes.py

```diff
@@ -45,7 +45,11 @@ def upload():
         file_format = excel_file.content_type
         if file_format != MIMETYPE.XLSX:
             error = ["The selected file must be an Excel file."]
-            current_app.logger.info("Incorrect file format uploaded")
+            current_app.logger.info(
+                f"Pre-validation error: {error}"
+                if Config.ENABLE_VALIDATION_LOGGING
+                else "Pre-validation error(s) found during upload"
+            )
```

Comment:
Actually - to help make sure we keep these messages consistent to make for easier analysis of the logs, we could pull them out as constants and then use `.format()` to inject the errors into them

Reply:
We can pull the log strings out as constants like:
```
PRE_VALIDATION_ERROR_LOG = "Pre-validation error: {error}"
PRE_VALIDATION_LOG = "Pre-validation error(s) found during upload"
```

and then use them like this:

```suggestion
            current_app.logger.info(
               PRE_VALIDATION_ERROR_LOG.format(error)
                if Config.ENABLE_VALIDATION_LOGGING
                else PRE_VALIDATION_LOG
            )
```

and we can reuse these later here

https://github.com/communitiesuk/funding-service-design-post-award-submit/blob/1467b9c5cbc9f10802f5a72429774d05790b663e/app/main/routes.py#L73-L77

like

```
 if Config.ENABLE_VALIDATION_LOGGING: 
     for pre_err in pre_errors: 
         current_app.logger.info(PRE_VALIDATION_ERROR_LOG.format(pre_err) 
 else: 
     current_app.logger.info(PRE_VALIDATION_LOG) 
```

Reply:
Using either .format() or f strings didn't work - it returned an error in the default.py file if the variable wasn't available before run time. I have just used string concatentation to pull the constants through to routes.py instead - though not sure if this is overly complex now

Reply:
The reason for my suggestion to pull them out was just to ensure that the messages are constant as they're used in multiple places in the code now - this way we can define it once and then reference that. If you think it's overly complex though then happy to approve without this change. 

If you do want to implement it though then I'd recommend these improvements:

I think the constants are best placed either in [const.py](https://github.com/communitiesuk/funding-service-design-post-award-submit/blob/SMD-361/feature/validation-error-logging/app/const.py) or just at the top of `routes.py` rather than in `Config` - as these are constants rather than anything configurable.

And if they're written like:

```
PRE_VALIDATION_ERROR_LOG = "Pre-validation error: {error}"
PRE_VALIDATION_LOG = "Pre-validation error(s) found during upload"
VALIDATION_ERROR_LOG = "Validation error: {error}"
VALIDATION_LOG = "Validation error(s) found during upload"
```

then you should be able to do

```
PRE_VALIDATION_ERROR_LOG.format(error=some_error)
VALIDATION_ERROR_LOG.format(error=some_error)
```

Happy to jump on a call if you want - just let me know